src/pages/execution/tabLoader.tsx

```diff
@@ -255,6 +263,106 @@ export default function tabLoader(tabBase: TabBase): TabData {
   };
 }
 
+function AlertComponent() {
+  const executionAndTasks = useContext(TileFactoryContext).executionAndTasks;
+  const { setSeverity } = useContext(TileFactoryContext);
+  const alerts = useMemo(() => {
+    const allAlerts: AlertItem[] = [];
+
+    rules.forEach((rule) => {
+      const ruleAlerts = rule(executionAndTasks);
+      allAlerts.push(...ruleAlerts);
+    });
+
+    return allAlerts;
+  }, [executionAndTasks]);
+
+  useEffect(() => {
+    const maxSeverity = findMaxSeverity(alerts, undefined);
+    setSeverity(maxSeverity);
+  }, [alerts, setSeverity]);
+
+  if (alerts.length === 0) {
+    return null;
+  }
+
+  return (
+    <Stack sx={{ margin: "15px" }} spacing={2}>
+      {alerts.map((alert, index) => (
+        <React.Fragment key={index}>{alert.component}</React.Fragment>
+      ))}
+    </Stack>
+  );
+}
+
+function SummaryTabHead() {
```

Comment:
Could we merge all of this (including findMaxSeverity etc)  into `Summary.tsx`? This way we can keep `tabLoader.tsx` generic as a router component.